strawberryfields/backends/fockbackend/circuit.py

```diff
@@ -763,6 +763,12 @@ def measure_homodyne(self, phi, mode, select=None, **kwargs):
             # Numpy also does not use the log probabilities
             probs = rho_dist.flatten().real
             probs /= np.sum(probs)
+
+            # Due to floating point precision error, values in the calculated probability distribution
+            # may have a very small negative value of -epsilon. The following sets
+            # these small negative values to 0.
+            probs[np.abs(probs) < 1e-16] = 0
```

Comment:
If e.g. a value in `probs` is `-1e-14` then this won't set it to zero, since you're just comparing the size of absolute values. Is this actually what you want? I would think that something like `probs[probs < 0] = 0` instead, but maybe I'm missing something.

Reply:
Ah, this is because I don't want to set all negative values to 0, I only want to set very small values (negative or positive) to zero.

This fix above will catch issues caused by floating point imprecision (where we expect the negative value to be in the range `-2*epsilon < p < 0`, where `epsilon ~ np.finfo(float).eps`), but if there is ever the case where `p = -0.2`, for example, than this indicates a systematic bug and I would prefer an exception is still raised 🙂

Reply:
Ok, I see. That makes sense. Thank you! :smile:

Reply:
This is more in line with that @thisac is suggesting: https://github.com/XanaduAI/thewalrus/blob/078837ebe86a4d39edb6d06b06b67ed6c8a1b606/thewalrus/quantum.py#L1091

I am happy with @josh146 solution, although am not sure if `1e-16` is the correct threshold value.

Reply:
The syntax here is really neat! :fire: 

`1e-16` seems pretty small indeed. Not sure about precision errors arising in these cases, but e.g. in PennyLane the tolerance being used is mostly `1e-10`.

Reply:
Oops, I might have made a typo here. I noticed when reproducing the bug that the negative probability values were always `2.220446049250313e-16`, which is machine epsilon. I can increase this to 1e-10, but it seems a bit arbitrary.

Reply:
That makes sense! Happy if the threshold is adjusted that way then :)

Reply:
```suggestion
            probs[np.abs(probs) < 1e-10] = 0
```